src/lib.rs

```diff
@@ -25,7 +25,7 @@ mod woff2 {
     fn drop(&mut self) {
       unsafe {
         if !self.inner.is_null() {
-          FreeMemoryOutput(self.inner);
+          FreeMemoryOutput(self);
```

Comment:
Based on your description, we may not need to use C++ code to free `Woff2MemoryOut` at all. We can simply use `std::mem::drop(Vec::from_raw_parts(self.data, self.length, self.length + 4096))` here.

Reply:
are you sure about that